gateway-service-session/src/main/java/org/apache/knox/gateway/service/session/SessionResource.java

```diff
@@ -66,15 +73,23 @@ public SessionInformation getSessionInformation() {
   }
 
   private String getBaseGatewayUrl(GatewayConfig config) {
-    return request.getRequestURL().substring(0,
-        request.getRequestURL().length() - request.getRequestURI().length()) +
-        "/" + config.getGatewayPath();
+    return request.getRequestURL().substring(0, request.getRequestURL().length() - request.getRequestURI().length()) + "/" + config.getGatewayPath();
   }
 
-  private String getLogoutPageUrl(GatewayConfig config) {
-    return getBaseGatewayUrl(config) +
-        "/knoxsso/knoxauth/logout.jsp?originalUrl=" + getBaseGatewayUrl(config) +
-        "/homepage/home";
+  private String getLogoutPageUrl(GatewayConfig config, String logoutPageProfile, String logoutPageTopologies) {
+    if (baseLogoutPageUrl == null) {
+      baseLogoutPageUrl = getBaseGatewayUrl(config) + "/knoxsso/knoxauth/logout.jsp?originalUrl=" + getBaseGatewayUrl(config) + "/homepage/home";
```

Comment:
Is the `knoxsso` part of the URL always fixed, or topology dependent?

Reply:
It's always fixed; I just made sure not to re-calculate this string every time. Nothing changed since previous versions.